test/contract/generatepin/generatePin.js

```diff
@@ -0,0 +1,103 @@
+                    expect(res.text).to.equal('');
+                    done();
+                });
+        });
+    });
+
+    describe('2. Missing Session-Id which should produce a 400 Bad Request', function () {
+        it('Returns HTTP 400 status', function (done) {
+            request(pinServiceUrl)
+                .get('')
+                .query({phoneNumber: '+447701111111'})
+                .expect(400)
+                .end(function (err, res) {
+                    if (err) {
+                        console.log('error raised: ', err);
+                    }
+                    expect(err).to.be.equal(null);
+                    done();
+                });
+        });
+    });
+
+    describe('3. Valid International Number', function () {
+        it('Returns HTTP 200 status and pin number', function (done) {
+            request(pinServiceUrl)
+                .get('')
+                .query({phoneNumber: '+61437112945'})
+                .set('Session-Id', '012233456789')
+                .expect(200)
+                .end(function (err, res) {
+                    if (err) {
+                        console.log('error raised: ', err);
+                    }
+                    expect(err).to.be.equal(null);
+                    expect(res.header).to.have.property('content-length').eq('6');
+                    expect(res.text).is.not.equal(null);
+                    done();
+                });
+        });
+    });
+
+    describe('4. Valid UK Number', function () {
+        it('Returns HTTP 200 status and pin number', function (done) {
+            request(pinServiceUrl)
+                .get('')
+                .query({phoneNumber: '+447535538319'})
+                .set('Session-Id', '012233456789')
+                .expect(200)
+                .end(function (err, res) {
+                    if (err) {
+                        console.log('error raised: ', err);
+                    }
+                    expect(err).to.be.equal(null);
+                    expect(res.header).to.have.property('content-length').eq('6');
+                    expect(res.text).is.not.equal(null);
+                    done();
+                });
+        });
+    });
+
+    describe('5. Valid UK Local Number', function () {
+        it('Returns HTTP 200 status and pin number', function (done) {
+            request(pinServiceUrl)
+                .get('')
+                .query({phoneNumber: '07535538319'})
+                .set('Session-Id', '012233456789')
+                .expect(200)
+                .end(function (err, res) {
+                    if (err) {
+                        console.log('error raised: ', err);
+                    }
+                    expect(err).to.be.equal(null);
+                    expect(res.header).to.have.property('content-length').eq('6');
+                    expect(res.text).is.not.equal(null);
+                    done();
+                });
+        });
+    });
+});
```

Comment:
Can you check the last 3 tests, they appear to be failing as they're returning a 500 status rather than a 200 status.

Reply:
The 500 errors were being generated on the business service becasue of missing or incorrect token values in the application.yml file. This was demo's with the correct values.